app/controllers/reservations_controller.rb

```diff
@@ -0,0 +1,54 @@
+class ReservationsController < ApplicationController
+  before_action :authenticate_user!
+
+
+  def new
+    @reservation = Reservation.new
+    render "reservation", locals: {list_id: params[:id]}
+  end
+
+  # aka "update" method (there's two)
+  def ship
+    @reservation = Reservation.find(params[:id])
+
+    if @reservation.donor_id == current_user.id || current_user.administrator?
+      @reservation.ship
+      @reservation.save
+      # TODO obviously not
+      redirect_to login_path
+    else
+      # TODO
+      redirect_to root_path, alert: "You are not authorized to access that page."
```

Comment:
Don't worry about checking for an admin - there are separate actions in the admin panel for admins to do this